python/ycm/vimsupport.py

```diff
@@ -465,17 +465,55 @@ def GetIntValue( variable ):
 # start and end are objects with line_num and column_num properties
 #
 # returns the delta (in characters) to the line length applied
-#
-# PreCondition: start and end are on the same line.
 def ReplaceChunk ( start, end, replacement_text, delta ):
```

Comment:
This function is now seriously complex. Any chance of splitting it up? Either way, with this increase in complexity, this will need _very_ good test coverage before we can trust it to stay working.